libs/shared/src/main/java/io/crate/common/collections/Maps.java

```diff
@@ -51,6 +52,32 @@ public static <K, V> Map<K, V> concat(Map<K, V> m1, Map<K, V> m2) {
         return Collections.unmodifiableMap(result);
     }
 
+    public static <K, V> Map<K, Set<V>> concatMultiMap(Map<K, Set<V>> m1, Map<K, Set<V>> m2) {
```

Comment:
How about changing this to something like `mergeMap(Map<K, V> m1, Map<K, V> m2, Function<V, V> mergeValue)`, could then be used with `mergeMap(m1, m2, Sets::union)` for the current use case and would be more flexible

Reply:
This is a good idea.